### PR TITLE
created: monster killer game

### PR DESCRIPTION
**Pull request by mmusk**

### CODE REVIEW

File: assets/scripts/app.js [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+const MONSTER_ATTACK_VALUE = 14;
+const HEAL_VALUE = 20;
+const MODE_ATTACK = 'ATTACK';
+const MODE_STRONG_ATTACK = 'STRONG_ATTACK';
+const LOG_EVENT_PLAYER_ATTACK = 'PLAYER_ATTACK';
+const LOG_EVENT_PLAYER_STRONG_ATTACK = 'PLAYER_STRONG_ATTACK';
+const LOG_EVENT_MONSTER_ATTACK = 'MONSTER_ATTACK';
+const LOG_EVENT_HEAL = 'HEAL';
+const LOG_EVENT_GAME_OVER = 'GAME_OVER';
+let battleLog = [];
```
Comment (Rehmatulla-Khan) [inline assets/scripts/app.js]: can we use `const` here ? @mmusk

File: assets/scripts/app.js [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    // } else if (ev === LOG_EVENT_PLAYER_ATTACK){
+    //     logEntry.target = 'Monster'
+    // } else if (ev === LOG_EVENT_PLAYER_STRONG_ATTACK){
+    //     logEntry = {
+    //         event: ev,
+    //         value: val,
+    //         target: 'Monster',
+    //         finalPlayerHealth: playerHealth,
+    //         finalMonsterHealth: monsterHelath
+    //     };
```
Comment (Rehmatulla-Khan) [inline assets/scripts/app.js]: remove comments @mmusk

File: assets/scripts/app.js [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+
+function healHandler(){
+    let healValue;
+    if (currentPlayerHealth >= choosenMaxLife - HEAL_VALUE){
+        alert("You can't heal more.")
+        healValue = choosenMaxLife - currentPlayerHealth;
+    } else{
+        healValue = HEAL_VALUE;
+    }
```
Comment (Rehmatulla-Khan) [inline assets/scripts/app.js:193]: can you use ternary operator here ? @mmusk

File: assets/scripts/app.js [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+const LOG_EVENT_PLAYER_ATTACK = 'PLAYER_ATTACK';
+const LOG_EVENT_PLAYER_STRONG_ATTACK = 'PLAYER_STRONG_ATTACK';
+const LOG_EVENT_MONSTER_ATTACK = 'MONSTER_ATTACK';
+const LOG_EVENT_HEAL = 'HEAL';
+const LOG_EVENT_GAME_OVER = 'GAME_OVER';
+let battleLog = [];
+
+
+function writeToLog (ev, val, playerHealth, monsterHelath){
+    let logEntry = {
```
Comment (faran4engg) [inline assets/scripts/app.js]: why let here ? can you think of better alternative!

File: assets/scripts/app.js [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+    if (isNaN(parsedValue) || parsedValue <=0){
+        throw{message: 'Invalid Input, enter a number!'};
+    }
+    return parsedValue;
+}
+
+try {
+    choosenMaxLife = getMaxLifeValue();
+} catch (error) {
+    console.log(error);
```
Comment (faran4engg) [inline assets/scripts/app.js]: we can use `console.warn` or `console.error` here

File: assets/scripts/app.js (hunk 3 above)
Comment (faran4engg) [reply assets/scripts/app.js:193]: i would prefer if..else because here its more than 2 statements

File: index.html [hunk 6]
```diff
@@ -0,0 +1,26 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <title>Monster Killer</title>
+    <link rel="stylesheet" href="./assets/styles/app.css" />
+    <script src="./assets/scripts/vendor.js" defer></script>
+    <script src="./assets/scripts/app.js" defer></script>
```
Comment (faran4engg) [inline index.html]: will the script like this delays the painting process ?

File: assets/scripts/app.js (hunk 3 above)
Comment (Rehmatulla-Khan) [reply assets/scripts/app.js:193]: yeah sure

Comment (mmusk) [reply assets/scripts/app.js:193]: Thanks @faran4engg

File: assets/scripts/app.js (hunk 2 above)
Comment (mmusk) [reply assets/scripts/app.js]: done @Rehmatulla-Khan

File: index.html (hunk 6 above)
Comment (mmusk) [reply index.html:10]: fixed @faran4engg

File: assets/scripts/app.js (hunk 1 above)
Comment (mmusk) [reply assets/scripts/app.js]: done @Rehmatulla-Khan

File: assets/scripts/app.js (hunk 5 above)
Comment (mmusk) [reply assets/scripts/app.js]: fixed @faran4engg

File: assets/scripts/app.js (hunk 2 above)
Comment (faran4engg) [inline assets/scripts/app.js]: How can this conversation is marked as resolved. I can still see the commented code. BTW, the reviewer should resolve the conversation 🙂

Comment (mmusk) [reply assets/scripts/app.js]: I'm using the web version of VS code I guess there's a glitch @faran4engg

**Review by Tanvir04khan — APPROVED** [review]

**Comment by Rehmatulla-Khan** [comment]
@mmusk I found a bug in UI when a player gets bonus life his health doesn't increase

**Review by Naufil-Shemle — APPROVED** [review]
Nice

File: assets/scripts/app.js [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+
+
+function reset(){
+    currentMonsterHealth = choosenMaxLife;
+    currentPlayerHealth = choosenMaxLife;
+    resetGame(choosenMaxLife);
+}
+
+
+function endRound(){
```
Comment (syedabdullahali) [inline assets/scripts/app.js:98]: `u can use here arrow function`

**Comment by mmusk** [comment]
> @mmusk I found a bug in UI when a player gets bonus life his health doesn't increase

fixed @Rehmatulla-Khan

**Review by Rehmatulla-Khan — APPROVED** [review]

**Review by Tauqeer-Ahmed-99 — APPROVED** [review]
Lgtm

File: assets/scripts/app.js (hunk 7 above)
Comment (mmusk) [reply assets/scripts/app.js:98]: I purposely didn't used it beacause I wasn't aware of it back then @syedabdullahali

**Review by faran4engg — APPROVED** [review]

**Review by anask601 — APPROVED** [review]

**Review by nehalskdev — APPROVED** [review]

**Pull request merged**